govcd/vapp_network_test.go

```diff
@@ -316,3 +316,65 @@ func (vcd *TestVCD) Test_UpdateNetworkNatRules(check *C) {
 	check.Assert(err, IsNil)
 	check.Assert(task.Task.Status, Equals, "success")
 }
+
+func (vcd *TestVCD) Test_UpdateNetworkStaticRoutes(check *C) {
+	vapp, networkName, vappNetworkConfig, err := vcd.prepareVappWithVappNetwork(check, "Test_UpdateNetworkStaticRoutes")
+	check.Assert(err, IsNil)
+
+	networkFound := types.VAppNetworkConfiguration{}
+	for _, networkConfig := range vappNetworkConfig.NetworkConfig {
+		if networkConfig.NetworkName == networkName {
+			networkFound = networkConfig
+		}
+	}
+	check.Assert(networkFound, Not(Equals), types.VAppNetworkConfiguration{})
+
+	desiredNetConfig := types.NetworkConnectionSection{}
+	desiredNetConfig.PrimaryNetworkConnectionIndex = 0
+	desiredNetConfig.NetworkConnection = append(desiredNetConfig.NetworkConnection,
+		&types.NetworkConnection{
+			IsConnected:             true,
+			IPAddressAllocationMode: types.IPAllocationModePool,
+			Network:                 "Test_UpdateNetworkStaticRoutes",
+			NetworkConnectionIndex:  0,
+		})
+
+	uuid, err := GetUuidFromHref(networkFound.Link.HREF, false)
+	check.Assert(err, IsNil)
+
+	result, err := vapp.UpdateNetworkStaticRouting(uuid, []*types.StaticRoute{&types.StaticRoute{Name: "test1",
+		Network: "192.168.2.0/24", NextHopIP: "192.168.2.15"}, &types.StaticRoute{Name: "test2",
```

Comment:
From my experiments at least NextHopIp must be related to network attached to avoid error:
<img width="754" alt="image" src="https://user-images.githubusercontent.com/15804230/85518177-a4da8780-b608-11ea-838d-fbce88b20220.png">

This test does pass now on one env I tested, but flagging just to be sure that these hard coded values don't break tests on envs with different IPs.

Reply:
Test now create own network, so should be fine.